packages/descendants/src/Highlight/reducer/makeHighlightReducerFunction.ts

```diff
@@ -0,0 +1,76 @@
+import { isDefined } from '@leafygreen-ui/lib';
+
+import {
+  Descendant,
+  DescendantsList,
+  getDescendantById,
+  getDescendantByIndex,
+} from '../../Descendants';
+import { getDeltaFromDirection } from '../utils/getDeltaFromDirection';
+import { getIndexFromPosition } from '../utils/getIndexFromPosition';
+import { getRelativeDescendant } from '../utils/getRelativeDescendant';
+
+import { HighlightReducerFunction } from './reducer.types';
+
+interface ReducerFunctionOptions<T extends HTMLElement> {
+  filter?: (d: Descendant<T>) => boolean;
+}
+
+/**
+ * Creates a new reducer function for closure for a given `descendants` value
+ */
+export const makeHighlightReducerFunction = <T extends HTMLElement>(
+  getDescendants: () => DescendantsList<T>,
+  options?: ReducerFunctionOptions<T>,
+): HighlightReducerFunction<T> => {
+  return (currentHighlight, action) => {
+    const descendants = getDescendants();
```

Comment:
because this is relative to what ever you pass in and not the canonical number of descendants, you should be able to filter on the component side which gets passed as the arg (line 23)

Reply:
See https://github.com/mongodb/leafygreen-ui/pull/2423#discussion_r1673122693